store/src/java/com/zimbra/cs/security/sasl/ZimbraAuthenticator.java

```diff
@@ -92,24 +93,29 @@ public ZimbraAuthenticator(AuthenticatorUser user) {
         }
 
         // make sure that the authentication account is valid
-        Account authAccount = prov.get(Key.AccountBy.name, authenticateId, at);
+        Account authAccount = prov.get(at.isAdmin() ? Key.AccountBy.adminName : Key.AccountBy.name, authenticateId, at);
         if (authAccount == null)
             return null;
 
         // make sure the auth token belongs to authenticatedId
         if (!at.getAccountId().equalsIgnoreCase(authAccount.getId()))
             return null;
-        
+
         // make sure the protocol is enabled for the user
         if (!isProtocolEnabled(authAccount, protocol)) {
             ZimbraLog.account.info("Authentication failed - %s not enabled for %s", protocol, authAccount.getName());
             return null;
         }
 
         // if necessary, check that the authenticated user can authorize as the target user
-        Account targetAcct = authorize(authAccount, username, AuthToken.isAnyAdmin(at));
-        if (targetAcct != null)
+        if (at.isAdmin() && username.equals(authenticateId) && username.equals(LC.zimbra_ldap_user.value())) {
```

Comment:
I think that instead of restricting this to LC.zimbra_ldap_user, this code should perform the same permission check as the one performed by FlushCache SOAP handler. Also, FlushCache SOAp handler should pass on the token of whoever is calling FlushCache. This way we have a continuous audit trail of who flushed the cache.